hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -2809,6 +2809,19 @@ func (r *HostedClusterReconciler) delete(ctx context.Context, req ctrl.Request,
 		return false, nil
 	}
 
+	var hostedControlPlaneList hyperv1.HostedControlPlaneList
```

Comment:
There must be only one HCP otherwise we might be hiding a bug.
Can we just:
```
	// There are scenarios where CAPI might not be operational e.g None Platform.
	// We want to ensure the HCP resource is deleted before deleting the Namespace.
	// Otherwise the CPO will be deleted leaving the HCP in a perpetual terminating state preventing further progress.
	hcp := controlplaneoperator.HostedControlPlane(controlPlaneNamespace, hc.Name)
	if err := r.Delete(ctx, hcp); err != nil {
		if !apierrors.IsNotFound(err) {
			return false, fmt.Errorf("error deleting Cluster: %w", err)
		}
	} else {
		r.Log.Info("Waiting for Hosted Control Plane deletion", "Name", hcp.Name, "Namespace", hcp.Namespace)
		return false, nil
	}
```

Reply:
Yea!!! We totally can will change